Perform OCSP requests manually

mirokuratczyk opened the pull request:
- Require revocation check to pass
- Require OCSP revocation check to pass
- Perform OCSP requests manually to ensure
   that they respect NSURLProtocol

mirokuratczyk committed:
Perform OCSP requests manually

- Require revocation check to pass
- Require OCSP revocation check to pass
- Perform OCSP requests manually to ensure
   that they respect NSURLProtocol

mirokuratczyk committed:
Integrate OCSPCache

Motivation:
OCSP requests are made in plaintext by
the system on iOS. These requests are
made outside of the tunnel.

Modifications:
Make OCSP requests manually through
the tunnel with OCSPCache.

Result:
OCSP requests cannot be intercepted
between the client and server.

efryntov commented on Podfile:
```diff
@@ -12,6 +12,8 @@ target "Psiphon Browser" do
 	#pod "InAppSettingsKit", :path => "../InAppSettingsKit"
 	pod 'PsiphonClientCommonLibrary', :git => "https://github.com/Psiphon-Inc/psiphon-ios-client-common-library.git", :commit => '2e53a34'
```
Not really relevant for this PR, but this pod entry is behind current https://github.com/Psiphon-Inc/psiphon-ios-client-common-library branch tip

mirokuratczyk replied:
Fixed in https://github.com/Psiphon-Inc/endless/pull/161/commits/83005f8d4fe502c849dc58799c26a9a3ecd2751f.

mirokuratczyk committed:
Update OCSPCache

Motivation:
OCSPCache now offers code for evaluating auth
challenges which provides control over how OCSP
requests are made.

Modifications:
- Remove JAHPSecTrustEvaluation
- Use OCSPAuthURLSessionDelegate

Result:
Auth challenges are now evaulated with code provided
by OCSPCache.

mirokuratczyk committed:
Updated common library version to 1.2.1

mirokuratczyk committed:
Update OCSPCache

- Bugfix in OCSPCache/0ebecf6

mirokuratczyk committed:
Update OCSPCache for NSURLSession reuse

Motivation:
Currently a new NSURLSession object is created for each
OCSP request. OCSPCache/c11ab4a now offers NSURLSession
reuse when making OCSP requests. This means the shared
NSURLSession object used by JAHP can be reused for all
OCSP requests.

Modifications:
- Update OCSPCache pod
- Use shared NSURLSession for making OCSP requests

Result:
All OCSP requests use a single NSURLSession object
which is shared with JAHP.
